datagouv/resource.py

```diff
@@ -170,9 +173,26 @@ def create_static(
         if is_communautary:
             url += "community/"
         logging.info(f"🆕 Creating '{payload['title']}' for {file_to_upload}")
-        r = self._client.session.post(url, files={"file": open(file_to_upload, "rb")})
-        r.raise_for_status()
-        metadata = r.json()
+        if from_url:
+            if file_name is None or mime is None:
+                raise ValueError(
+                    "When uploading from a URL, file_name and mime arguments are required."
+                )
+            with self._client.session.stream("GET", file_to_upload) as streamed:

```

Comment:
Could we open/close the session stream without a `with` context in order to mutualise the POST logic with just a different `files` dict?